java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -1925,6 +1940,130 @@ public Table gather(ColumnVector gatherMap, boolean checkBounds) {
     return new Table(gather(nativeHandle, gatherMap.getNativeView(), checkBounds));
   }
 
+  private GatherMap[] buildJoinGatherMaps(long[] gatherMapData) {
+    long bufferSize = gatherMapData[0];
```

Comment:
nit: If the size of the array is fixed, would it be faster to pass down the array instead of allocating it in JNI? It is not a big deal, but I think that was one of the conclusions from previous tests.

Reply:
If that's true, then we should apply this logic to all of the `Table` native methods that are returning `long[]`, since the code should know the number of array entries for almost all of them up-front.  I think we should keep this consistent with every other native `Table` method for now and do the 'allocate arrays in Java first' change as a followup.

Reply:
I probably misremembered it.